hex-literal/src/lib.rs

```diff
@@ -83,16 +83,28 @@ impl TokenTreeIter {
     }
 
     fn next_hex_val(&mut self) -> Option<u8> {
+        let mut incomment = false;
         loop {
             let v = match self.buf.get(self.pos) {
                 Some(&v) => v,
                 None => return None,
             };
             self.pos += 1;
+            if incomment {
+                match v {
+                    b'\n' => { incomment = false; }
+                    _ => { /* nothing */ }
```

Comment:
I think this `match` can be replaced with a single line:
```rust
incomment = v == b'\n';
```

It (and your variant) may be somewhat incorrect for characters which contain `\n` in their UTF-8 encoding, but I guess it's better to simply forbid non-ASCII comments for now.